source/common/conn_pool/conn_pool_base.cc

```diff
@@ -34,6 +34,10 @@ void ConnPoolImplBase::destructAllConnections() {
 }
 
 bool ConnPoolImplBase::shouldCreateNewConnection() const {
+  // If the host is unhealthy, don't make it do extra work.
+  if (host_->health() != Upstream::Host::Health::Healthy) {
```

Comment:
I could go either way, but I intentionally did this for !Healthy rather than Unhealthy for a couple of reasons.  Late connection fetching could result in up to date health information if there were a change in health status and if lookahead prefetching will avoid degraded hosts, I think prefetching may be overenthusiastic given they're likely to be avoided.  Basically I think when hosts are not healthy, the LB is less predictable so I lean towards taking the latency hit over a CPU hit but it's easy enough to go the other way if either of your care.

Reply:
I would suggest a TODO to consider ideal prefetch behavior for degraded endpoints.  Limiting this version to healthy endpoints seems fine.

Reply:
Sounds reasonable. Maybe add some of that explanation as a code comment?

Reply:
I'd lean against a TODO as I only know of one user of degraded upstream and don't know if they need prefetching.  I'll note it could be extended and comment why we don't by default.  SG?

Reply:
Good enough for me. @antoniovicente ?